libkode/printer.cpp

```diff
@@ -956,16 +951,47 @@ void Printer::printImplementation( const File &file, bool createHeaderInclude )
     filename.prepend( d->mOutputDirectory + '/' );
 
   QFile implementation( filename );
-  if ( !implementation.open( QIODevice::WriteOnly ) ) {
-    qWarning( "Can't open '%s' for writing.", qPrintable( filename ) );
-    return;
+  printCodeIntoFile(out, &implementation);
+}
+
+void Printer::printCodeIntoFile(const Code &code, QFile *file)
+{
+  QString outText = code.text();
+  bool identical = true;
+  QTextStream fileStream( file );
+  if (QFile::exists(file->fileName())) {
+    if ( !file->open( QIODevice::ReadOnly ) ) {
+      qWarning( "Can't open '%s' for reading.", qPrintable( file->fileName() ) );
+      return;
+    }
+    QTextStream codeStream( &outText );
+    QString fileLine, outLine;
+    while (fileStream.readLineInto(&fileLine) && codeStream.readLineInto(&outLine)) {
+      if (fileLine != outLine) {
+        identical = false;
+        break;
+      }
+    }
+    identical = fileStream.atEnd() && codeStream.atEnd();
+    file->close();
+  } else {
+    identical = false;
   }
 
-  QTextStream h( &implementation );
+  if (!identical) {
+    if ( !file->open( QIODevice::WriteOnly ) ) {
+      qWarning( "Can't open '%s' for writing.", qPrintable( file->fileName() ) );
+      return;
+    }
 
-  h << out.text();
+    fileStream << outText;
```

Comment:
I find it dangerous to reuse the same fileStream (which thinks it's at the end of the file) after changing its underlying QFile object. Even if this works today, changes in Qt might break this unusual use case.

I would just create a different QTextStream for reading and for writing.
And set both to UTF-8, while at it.

Reply:
Fixed.